rpc/core/env.go

```diff
@@ -35,12 +36,16 @@ const (
 
 var (
 	// set by Node
+	mut = &sync.Mutex{}
 	env *Environment
 )
 
-// SetEnvironment sets up the given Environment.
-// It will race if multiple Node call SetEnvironment.
+// SetEnvironment sets up the given Environment. The env global var that this
+// function modifies is protected by a sync.Once, so multiple calls within the
+// same process will not be effective.
 func SetEnvironment(e *Environment) {
+	mut.Lock()
```

Comment:
[Non Blocking][Question][Not urgent]
If I understand right, this would remove race conditions. However, I checked where the environment is used, and apparently it is used for a loooot, from querying blocks to mempool etc.
I'm curious in why there will be a race condition?
I'm thinking of two possibilities: 
- This variable is not set to system environment variables, but only the global scope of the runtime. This means that no race condition should happen, as there is no reason for two full nodes, in separate processes, to share the same runtime.
- This variable is setting system environment variables. And, in this case, adding a mutex here will not fix the issue as multiple nodes will need different configurations. Thus, we will need to have a way to actually separate the environment for both.

Reply:
basically only in celestia-node during the swamp tests as we spin up many tendermint nodes in the same process

Reply:
Aaah, understandable. And, these nodes have different configurations? Or, the same configuration?

Reply:
Rarely they have different configurations and usually they are the same.